core/dispatch.c

```diff
@@ -874,6 +874,17 @@ dispatch_enter_dynamorio(dcontext_t *dcontext)
         if (get_at_syscall(dcontext))
             handle_post_system_call(dcontext);
 
+#ifdef X86
+        /* If the next basic block starts at a debug register value,
+         * we fire a single step exception before getting to the basic block. */
```

Comment:
Wait, I don't follow your response: my comment was just verifying that we do not want to execute any instrumentation for this instruction, which I think is correct.  The client will not see a bb with this instruction until after the fault is raised.